src/main/java/com/wavefront/sdk/jersey/WavefrontJerseyFilter.java

```diff
@@ -318,4 +386,89 @@ private AtomicInteger getGaugeValue(MetricName metricName) {
       put("jersey" + ".resource.method", finalMethodName);
     }};
   }
+
+  private SpanContext parentSpanContext(ContainerRequestContext requestContext) {
+    Span activeSpan = tracer.activeSpan();
+    if (activeSpan != null) {
+      return activeSpan.context();
+    } else {
+      return tracer.extract(
+              Format.Builtin.HTTP_HEADERS,
+              new ServerHeadersExtractTextMap(requestContext.getHeaders())
+      );
+    }
+  }
+
+  private void decorateRequest(ContainerRequestContext requestContext, Span span) {
+    Tags.COMPONENT.set(span, JERSEY_SERVER_COMPONENT);
+    Tags.HTTP_METHOD.set(span, requestContext.getMethod());
```

Comment:
We need more span tags here. Talk to @srujann about it.

Reply:
@song-hao  these are the custom tags you can add.

```
component    jersey-server
span.kind    server
http.method    POST
http.url    http://localhost:50050/make/foo
http.status_code    200
error    true
jersey.path    "/make/{styleId}"
jersey.resource.class    com.wavefront.ShoppingWebResource
```